### PR TITLE
Populate Tvmult and Tvmult_add of an inverse_operator correctly

### LINKED ISSUE #2008: Error in compound LinearOperator [A^{-1}]^{T}
This [gist](https://gist.github.com/jppelteret/1872cd0b78477b6e3258) demonstrates some non-deterministic bug in `LinearOperator` (or `PackagedOperation`) which was mentioned in #1956 . Incorrect results *may* arise when performing computations with the transpose of the inverse of a matrix, while using the inverse of a transpose matrix appears to work as expected.

### PR DESCRIPTION
**Pull request by tamiko**
Fix a typo in linear_operator.h: The transpose operation function-object
Tvmult was accidentally overwritten by Tvmult_add (and the latter left
uninitialized).

Closes #2008

### CODE REVIEW

File: doc/news/changes.h [hunk 1]
```diff
@@ -526,6 +526,12 @@ inconvenience this causes.
 
 
 <ol>
+  <li> Fixed: inverse_operator() now populates <code>Tvmult</code> and
+  <code>Tvmult_add</code> correctly.
+  <br>
+  (Jean-Paul Pelteret, David Wells, Matthias Maier, 2015/12/30)
+  </li>
```
Comment (bangerth) [inline doc/news/changes.h]: Is this a change against 8.3, or was all of this introduced only post 8.3? In the latter case no entry is needed here.

Comment (tamiko) [inline doc/news/changes.h]: > Is this a change against 8.3 [...]?

Yes, it is.

**Comment by tjhei** [comment]
How come an uninitialized functions gave the wrong result instead of failing hard? Is there a way to change that? (patch looks good)

**Comment by tamiko** [comment]
@tjhei: ```Tvmult_add``` was never used (doing so would have lead to an exception). The bug @jppelteret encountered is caused by ```Tvmult``` being overwritten by ```Tvmult_add```.

**Comment by jppelteret** [comment]
@tamiko Thanks for getting to the bottom of this! I thought that the asserts in the base class constructor (~line 148) should have been triggered in such a scenario? I definitely looked over this line a couple of times...

**Comment by tamiko** [comment]
@jppelteret Unfortunately, no. There is no way how we can guard against accidentally overwriting the wrong function object...

**Pull request merged**